src/main/java/cards/CardChecker.java

```diff
@@ -71,6 +77,8 @@ private boolean isStreet(List<CardModel> cardModels) {
         return isStreet;
     }
 
+    // CR: in getCombination() you are receiving a List and not a set?
+    // is there a reason for the difference?
```

Comment:
I'm receiving a List. Than it was grouped to Map [Rank -> quantity of card in list]. The was created to iterate over this map.

Reply:
my question is if there is a reason that in getCombination() you are receiving a List, and in isFourOfKind() you are receiving a Set (of a Map).
Both list and set receive a single type (unlike map) and are iterated in the same way.
is there a value to receive a List in one place and Set in another?

Reply:
isFourOfKind() should receive a Map [Rank -> quantity of card in list]. A Set should be created, to iterate through it. Actually, three is no need  to create this Set outside of isFourOfKind() method scope but it should been created in any case for iteration process.